### DISCUSSION TITLE
How to map typedef array

### DISCUSSION THREAD

**Discussion by luxingxiao** [bytedeco/javacpp#634] [Q&A] [answered]
C language header:
`typedef uint8_t DDS_XTypes_NameHash[4];`

**Comment by saudet** [comment, marked as answer]
It's typically easiest to consider that the same as a `char*`.

**Comment by luxingxiao** [comment]
:(

.put(new Info("DDS_XTypes_NameHash").cppTypes("char*"))

error: incompatible types in assignment of ‘char*’ to ‘DDS_XTypes_NameHash’ {aka ‘unsigned char [4]’}
 4293 |     ptr->name_hash = (char*)ptr0;
      |     ~~~~~~~~~~~~~~~^~~~~~~~~~~~~
error: ISO C++ forbids casting to an array type ‘DDS_XTypes_QualifiedTypeName’ {aka ‘char [257]’}
 8310 |     DDS_XTypes_QualifiedTypeName rval = (DDS_XTypes_QualifiedTypeName)ptr->annotation_name;

**Reply by saudet** [reply]
We can't cast a pointer to an array, no. In a case like that, we might need to write some custom Java code like this:
https://github.com/bytedeco/javacpp/wiki/Mapping-Recipes#mapping-a-declaration-to-custom-code

**Reply by saudet** [reply]
Try to annotate with `@MemberGetter` like this:
https://github.com/bytedeco/javacpp/wiki/Mapping-Recipes#mapping-a-declaration-to-custom-code
